Source/Core/DolphinQt/NetPlay/NetPlayDialog.cpp

```diff
@@ -1019,10 +1018,13 @@ bool NetPlayDialog::IsRecording()
 std::string NetPlayDialog::FindGame(const std::string& game)
 {
   std::optional<std::string> path = RunOnObject(this, [this, &game] {
-    for (int i = 0; i < m_game_list_model->rowCount(QModelIndex()); i++)
+    if (m_game_list_model)
```

Comment:
Is it even necessary to `RunOnObject` here? Or is there a chance that `m_game_list_model` would test ok before the call but fail inside?
Thinking about whether `if (!m_game_list_model) return "";` could be outside the lambda, _before_ calling `RunOnObject`

Reply:
Good question. `NetPlayDialog::FindGame()` is a public function. Technically, it could be invoked from a non-UI thread, therefore, `RunOnObject()` is required; I don't think we can remove it. Without `RunOnObject()`, it wouldn't be safe to access UI components (i.e. `m_game_list_model` in this case). At least I think that was the intention of the original writer (spycrap).

So, I think we need `if (m_game_list_model)` inside the lambda. We could also have it outside the lambda as you suggested, but it's unlikely `m_game_list_model` will be `nullptr` at that time. The user would have to start a new NetPlay game and close Dolphin quickly in order to hit the sweet gap in which the GameListModel could be destroyed before finding the game in the list. It's extremely unlikely if at all possible, and, for correctness, we would still need to check inside the lambda to be safe.